packages/server/src/restaurants/database/RestaurantSchema.ts

```diff
@@ -1,15 +1,111 @@
-import { Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
 import { Exclude, Expose } from 'class-transformer';
 import { ObjectId } from 'mongodb';
 import { Document } from 'mongoose';
 
+import { Address } from '../../addresses/database/AddressSchema';
+
 type RestaurantDocument = Restaurant & Document<ObjectId>;
 
+enum CuisineTypes {
+  American = 'amerykańska',
+  Asian = 'azjatycka',
+  European = 'europejska',
+  Arab = 'arabska',
+  Chinese = 'chińska',
+  French = 'francuska',
+  Georgian = 'gruzińska',
+  Greek = 'grecka',
+  Indian = 'indyjska',
+  Italian = 'włoska',
+  Japanese = 'japońska',
+  Jewish = 'żydowska',
+  Korean = 'koreańska',
+  Lebanese = 'libańska',
+  Mediterranean = 'śródziemnomorska',
+  Mexican = 'meksykanśka',
+  Polish = 'polska',
+  Thai = 'tajska',
+  Turkish = 'turecka',
+  Vietnamese = 'wietnamska',
+}
+
+enum RestaurantTags {
+  FastFood = 'fast food',
+  Vegan = 'wegańska',
+  Vegetarian = 'wegetariańska',
+  Cafe = 'kawiarnia',
+  Healthy = 'zdrowa',
+  StreetFood = 'street food',
+  Desserts = 'desery',
+  Pizza = 'pizza',
+  Burgers = 'burgery',
+  Sushi = 'sushi',
+  Kebab = 'kebab',
+  GlutenFree = 'gluten free',
+}
+
+const RESTAURANT_CONSTANTS = Object.freeze({
+  NAME: Object.freeze({
+    MIN_LENGTH: 1,
+    MAX_LENGTH: 50,
+  }),
+  BANK_ACCOUNT_NUMBER: Object.freeze({
+    REGEX: /\W*\d{8,19}\b/,
+  }),
+  PHONE_NUMBER: Object.freeze({
+    REGEX: /(?<!\w)(\(?(\+|00)?48\)?)?[ -]?\d{3}[ -]?\d{3}[ -]?\d{3}(?!\w)/,
+  }),
```

Comment:
Mamy już [odpowiedni dekorator do walidacji numerów telefonu](https://github.com/CodersCamp2021-HK/CodersCamp2021.Project.Fullstack/blob/e1677f6f682a9f695ff03e03ad6d5d5f58ec5dca/packages/server/src/auth/api/decorators/ApiPhoneNumberProperty.ts#L6), lepiej będzie z niego skorzystać.

Reply:
Chodziło Ci może o [PHONE_NUMBER](https://github.com/CodersCamp2021-HK/CodersCamp2021.Project.Fullstack/blob/e1677f6f682a9f695ff03e03ad6d5d5f58ec5dca/packages/server/src/auth/shared/Constants.ts#L17)? Bo dekorator to jak rozumiem dopiero w `RestaurantDto.ts`

Reply:
No, w sumie oznaczyłem słaby fragment kodu. RegExp ten który już mamy żeby nie było rozbieżności (słabo jakby dany numer można było wprowadzić w jednym miejscu, a w drugim już nie) w schema + dekorator w DTO będzie ok.